packages/garbo/src/tasks/freeFight.ts

```diff
@@ -0,0 +1,614 @@
+  retrieveItem,
+  runChoice,
+  runCombat,
+  Skill,
+  use,
+  useSkill,
+  visitUrl,
+  weaponHands,
+} from "kolmafia";
+import {
+  $class,
+  $effect,
+  $familiar,
+  $item,
+  $items,
+  $location,
+  $monster,
+  $phyla,
+  $skill,
+  ChateauMantegna,
+  clamp,
+  CombatLoversLocket,
+  Delayed,
+  ensureEffect,
+  get,
+  have,
+  maxBy,
+  sum,
+  TunnelOfLove,
+  undelay,
+  uneffect,
+  Witchess,
+} from "libram";
+import { acquire } from "../acquire";
+import { GarboStrategy, Macro } from "../combat";
+import { globalOptions } from "../config";
+import { garboValue } from "../garboValue";
+import { freeFightOutfit } from "../outfit";
+import { GarboTask } from "./engine";
+
+type GarboFreeFightTask = GarboTask & {
+  combatCount: () => number;
+  tentacle: boolean; // if a tentacle fight can follow
+};
+
+const DEFAULT_FREE_FIGHT_TASK = {
+  // GarboTask
+  combat: new GarboStrategy(Macro.basicCombat()),
+  outfit: freeFightOutfit,
+  spendsTurn: false,
+  // GarboFreeFightTask
+  combatCount: () => 1,
+};
+
+function bestWitchessPiece() {
+  return maxBy(Witchess.pieces, (monster) =>
+    sum(itemDropsArray(monster), ({ drop }) => garboValue(drop)),
+  );
+}
+
+const isFree = (monster: Monster) => monster.attributes.includes("FREE");
+const valueDrops = (monster: Monster) =>
+  sum(itemDropsArray(monster), ({ drop, rate, type }) =>
+    !["c", "0", "p"].includes(type) ? (garboValue(drop, true) * rate) / 100 : 0,
+  );
+const locketMonster = () => CombatLoversLocket.findMonster(isFree, valueDrops);
+const locketsToSave = () =>
+  CombatLoversLocket.availableLocketMonsters().includes(
+    $monster`Knob Goblin Embezzler`,
+  )
+    ? 1
+    : 0;
+
+const maxSealsAvailable = () =>
+  retrieveItem(1, $item`Claw of the Infernal Seal`) ? 10 : 5;
+
+function sealsAvailable(): number {
+  const max = maxSealsAvailable();
+  const available = guildStoreAvailable()
+    ? Infinity
+    : Math.floor(availableAmount($item`seal-blubber candle`) / 3);
+  return Math.min(max, available);
+}
+
+const stunDurations = new Map<Skill | Item, Delayed<number>>([
+  [$skill`Blood Bubble`, 1],
+  [
+    $skill`Entangling Noodles`,
+    () =>
+      myClass() === $class`Pastamancer` && !have($skill`Shadow Noodles`)
+        ? 1
+        : 0,
+  ],
+  [$skill`Frost Bite`, 1],
+  [$skill`Shadow Noodles`, 2],
+  [
+    $skill`Shell Up`,
+    () => {
+      if (myClass() !== $class`Turtle Tamer`) return 0;
+      for (const [effect, duration] of new Map([
+        [$effect`Glorious Blessing of the Storm Tortoise`, 4],
+        [$effect`Grand Blessing of the Storm Tortoise`, 3],
+        [$effect`Blessing of the Storm Tortoise`, 2],
+      ])) {
+        if (have(effect)) return duration;
+      }
+      return 0;
+    },
+  ],
+  [$skill`Soul Bubble`, () => (mySoulsauce() >= 5 ? 2 : 0)],
+  [$skill`Summon Love Gnats`, 1],
+  [$item`Rain-Doh blue balls`, 1],
+]);
+
+const FreeFightTasks: GarboFreeFightTask[] = [
+  {
+    ...DEFAULT_FREE_FIGHT_TASK,
```

Comment:
I think there are some clever typescript ways of doing this wihtout having hte same ... in every entry